File: algorithms/1_two_pointers/5_merge_sorted_array/mergeSortedArray.py
from typing import List

def merge(nums1: List[int], m: int, nums2: List[int], n: int) -> None:
    """
    Do not return anything, modify nums1 in-place instead.
    """
    pointer_1 = m-1
    pointer_2 = n-1

    for i in range(len(nums1)-1,-1,-1):
        if pointer_1>=0 and pointer_2>=0 and nums1[pointer_1] >= nums2[pointer_2]:
            nums1[i]=nums1[pointer_1]
            pointer_1=pointer_1-1
        else:
            if pointer_2>=0:
                nums1[i]=nums2[pointer_2]
                pointer_2=pointer_2-1
            else:
                break

test = [1,2,3,0,0,0]
merge(test,3,[2,5,6],3)
assert test == [1,2,2,3,5,6]
print('test passed')

File: algorithms/1_two_pointers/5_merge_sorted_array/mergeSortedArray.js
'use strict';

function mergeSortedArray(nums1, m, nums2, n){


  let pointer_1 = m-1, pointer_2 = n-1;

  for (let i = nums1.length -1; i >= 0; i --){

    if (pointer_1>=0 && pointer_2>=0 && nums1[pointer_1] >= nums2[pointer_2]){

      nums1[i] = nums1[pointer_1];
      pointer_1-=1;

    } else {
      if(pointer_2>=0){
        nums1[i] = nums2[pointer_2];
        pointer_2-=1;
      } else {
        break;
      }
    }
  }
}

module.exports = mergeSortedArray;
